acquire/utils.py

```diff
@@ -9,9 +10,91 @@
 import traceback
 from enum import Enum
 from pathlib import Path
-from typing import Dict, List, Optional
+from stat import S_IRGRP, S_IROTH, S_IRUSR
+from typing import Dict, List, Optional, Union
 
-from acquire.outputs import OUTPUTS
+import acquire.outputs
+from dissect.target.helpers.fsutil import TargetPath
+from dissect.util.stream import AlignedStream
+
+
+class VolatileAlignedStream(AlignedStream):
+    """A volatile streaming class to handle files that live in volatile filesystem. Such as procfs or sysfs.
+    Handeling various edge-cases and OSErrors encountered within these filesystems.
+
+    Set max_filesize to 0 or `None` to remove the maximum filesize limitation.
+
+    Args:
+        path: The path to the file to acquire a VolatileStream of
+        string_flags: String flags, such as 'r' or 'rb' to open the file with.
+        flags: Integer flags. such as `O_RDONLY` or `O_NONBLOCK` to open the file with.
+        max_filesize: Set the maximum file size for streaming. Set to 0 or `None` to remove the maximum.
+    """
+
+    def __init__(
+        self,
+        path: Union[str, TargetPath],
+        string_flags: str = "rb",
+        flags: int = os.O_RDONLY | os.O_NONBLOCK | os.O_NOATIME,
+        max_filesize=5242880,  # 5mb
+    ):
+
+        self.max_filesize = max_filesize
+        self.fh = open(path, string_flags)
+        self.fd = self.fh.fileno()
+
+        fcntl.fcntl(self.fd, fcntl.F_SETFL, flags)
+
+        self.pos = 0
+        self.buf = b""
+
+        super().__init__()
```

Comment:
```suggestion
        super().__init__(max_filesize)
```
Also, if you were to do this, I believe the check you currently do in the loop is done for you by `AlignedStream`.

I believe you should be able to just do something like this:

```python
class VolatileStream(AlignedStream):
    def __init__(self, path: Path, mode: str = "rb", flags: int = os.O_RDONLY | os.O_NONBLOCK | os.O_NOATIME, size: int = 1024 * 1024 * 5):
        self.fh = path.open(mode)
        self.fd = self.fh.fileno()
        fcntl.fcntl(self.fd, fcntl.F_SETFL, flags)

        st_mode = os.fstat(self.fd).st_mode
        write_only = (st_mode & (S_IRUSR | S_IRGRP | S_IROTH)) == 0

        super().__init__(0 if write_only else size)

    def seek(self, pos: int, whence: int = io.SEEK_SET) -> int:
        raise UnsupportedOperation("VolatileStream is not seekable")

    def seekable(self) -> bool:
        return False

    def _read(self, offset: int, length: int) -> bytes:
        return os.read(self.fd, min(length, self.size - offset))
```